core/src/main/java/com/microsoft/applicationinsights/internal/persistence/LocalFileLoader.java

```diff
@@ -0,0 +1,110 @@
+package com.microsoft.applicationinsights.internal.persistence;
+
+import com.microsoft.applicationinsights.TelemetryChannel;
+import com.microsoft.applicationinsights.internal.config.connection.EndpointProvider;
+import com.microsoft.applicationinsights.internal.util.ThreadPoolUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static com.microsoft.applicationinsights.internal.persistence.PersistenceHelper.TEMPORARY_FILE_EXTENSION;
+
+/**
+ * This class manages loading a list of {@link ByteBuffer} from the disk.
+ */
+public class LocalFileLoader {
+
+    private static final Logger logger = LoggerFactory.getLogger(LocalFileLoader.class);
+    private static final long INTERVAL = TimeUnit.SECONDS.toSeconds(30); // send persisted telemetries from local disk every 30 seconds.
+    private static final ScheduledExecutorService scheduledExecutor = Executors.newSingleThreadScheduledExecutor(ThreadPoolUtils.createDaemonThreadFactory(LocalFileLoader.class));
+    private static TelemetryChannel telemetryChannel;
+    private static final LocalFileLoader INSTANCE = new LocalFileLoader();
+
+    /**
+     * Track a list of active filenames persisted on disk.
+     * FIFO (First-In-First-Out) read will avoid an additional sorting at every read.
+     */
+    private static final Queue<String> PERSISTED_FILES_QUEUE = new ConcurrentLinkedDeque<>();
```

Comment:
typically just constants use UPPER_CASE names

(see https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names for the definition of "but what is a constant?" that otel follows)

```suggestion
    private static final Queue<String> persistedFilesQueue = new ConcurrentLinkedDeque<>();
```